skrub/_joiner.py

```diff
@@ -12,20 +11,23 @@
 from sklearn.utils.validation import check_is_fitted
 
 from skrub import _join_utils, _matching, _utils
-from skrub._dataframe._namespace import is_pandas, is_polars
+from skrub import _selectors as s
+from skrub._check_input import CheckInputDataFrame
+from skrub._dataframe import _common as ns
 from skrub._datetime_encoder import DatetimeEncoder
+from skrub._to_str import ToStr
 
-from . import _selectors as s
 from ._wrap_transformer import wrap_transformer
 
 
-def _as_str(column):
-    return column.fillna("").astype(str)
+def _as_str(col):
+    return ToStr().fit_transform(col)
 
 
 DEFAULT_STRING_ENCODER = make_pipeline(
     FunctionTransformer(_as_str),
```

Comment:
```suggestion
    ToStr(),
```

and you can get rid of the function `_as_str`